packages/contracts-bedrock/src/cannon/interfaces/IPreimageOracle.sol

```diff
@@ -47,4 +47,19 @@ interface IPreimageOracle {
     /// @param _partOffset The offset of the preimage to read.
     /// @param _preimage The preimage data.
     function loadKeccak256PreimagePart(uint256 _partOffset, bytes calldata _preimage) external;
+
+    /// @notice Resets the caller's large pre-image metadata in preparation for beginning the absorption of a new
+    ///         large keccak256 pre-image.
+    /// @param _offset The offset of the preimage part to load into the oracle during absorbtion of the preimage.
+    /// @param _claimedSize The claimed size of the preimage.
+    function initLargeKeccak256Preimage(uint128 _offset, uint64 _claimedSize) external;
```

Comment:
Given this API, we'd need to design the challenger to ensure it only ever uploaded one large pre-image at a time. Currently it treats games as completely independent so could wind up trying to upload two different large preimages for different games at the same time and they'd conflict with each other.

We can make this work in the challenger - it's not particularly difficult, but it is potentially error prone.  I wonder if it's worth specifying the claimed hash of the data in init and using that to allow a single send to upload two different preimages in parallel.  We don't necessarily need to verify the final hash matches the claimed hash though it could be a useful validation.

Reply:
Yeah - great point. It's trivial to add a UUID to the mapping to allow for the challenger to be posting multiple preimages at one time, so will do that.